src/main/kotlin/com/hypto/iam/server/security/Authentication.kt

```diff
@@ -48,8 +49,9 @@ enum class TokenType(val type: String) {
 /** Class which stores the token credentials sent by the client */
 data class TokenCredential(val value: String?, val type: TokenType?) : Credential
 abstract class IamPrincipal : Principal {
-    abstract val organization: String
     abstract val tokenCredential: TokenCredential
+    abstract val organization: String
+    abstract val issuer: String
```

Comment:
Sure, will convert to `interface`, because, `val` in `abstract class` with initialization is considered to be `final`, thereby, achieving the same goal